Assets/BossRoom/Scripts/Server/Net/ServerGameNetPortal.cs

```diff
@@ -233,23 +247,75 @@ private void ApprovalCheck(byte[] connectionData, ulong clientId, MLAPI.NetworkM
             var connectionPayload = JsonUtility.FromJson<ConnectionPayload>(payload); // https://docs.unity3d.com/2020.2/Documentation/Manual/JSONSerialization.html
             int clientScene = connectionPayload.clientScene;
 
-            m_ClientSceneMap[clientId] = clientScene;
+            //a nice addition in the future will be to support rejoining the game and getting your same character back. This will require tracking a map of the GUID
+            //to the player's owned character object, and cleaning that object on a timer, rather than doing so immediately when a connection is lost. 
+            Debug.Log("Host ApprovalCheck: connecting client GUID: " + connectionPayload.clientGUID);
 
-            //TODO: GOMPS-78. We'll need to save our client guid so that we can handle reconnect.
-            Debug.Log("host ApprovalCheck: client guid was: " + connectionPayload.clientGUID);
+            ConnectStatus gameReturnStatus = ConnectStatus.Success;
 
-            //Populate our dictionaries with the playerData
-            m_ClientIDToGuid[clientId] = connectionPayload.clientGUID;
-            m_ClientData[connectionPayload.clientGUID] = new PlayerData(connectionPayload.playerName, clientId);
+            //Test for Duplicate Login. 
+            if( m_ClientData.ContainsKey(connectionPayload.clientGUID))
+            {
+                if( Debug.isDebugBuild )
+                {
+                    Debug.Log($"Client GUID {connectionPayload.clientGUID} already exists. Because this is a debug build, we will still accept the connection");
+                    while( m_ClientData.ContainsKey(connectionPayload.clientGUID)) { connectionPayload.clientGUID += "_Secondary"; }
+                }
+                else
+                {
+                    ulong oldClientId = m_ClientData[connectionPayload.clientGUID].m_ClientID;
+                    StartCoroutine(CoroDisconnectClient(oldClientId, ConnectStatus.LoggedInAgain));
+                }
+            }
 
-            //TODO: GOMPS-79 handle different error cases.

```

Comment:
this todo is still valid, should keep it

Reply:
done, although I moved the fixme to ClientGameNetPortal, because the application-layer failures ARE being handled. The only part of that ticket that is undone is handling transport failures more gracefully (that is what is waiting on MLAPI).